File: c++/0083-remove-duplicates-from-sorted-list.cpp
class Solution {
public:
    ListNode* deleteDuplicates(ListNode* head) {
      ListNode * fast = head;
      ListNode * slow = head;
      
      while(slow != NULL)
      {
        while(fast != NULL && slow->val == fast->val)
          fast = fast -> next;
        
        slow->next = fast;
        slow = slow -> next;
      }
      
      return head;
        
    }
};

File: kotlin/0083-remove-duplicates-from-sorted-list.kt
class Solution {
    fun deleteDuplicates(head: ListNode?): ListNode? {
        var dummy = ListNode(Integer.MAX_VALUE, head)
        var prev = dummy
        var current = head
        
        while(current != null) {
            if(current.`val` == prev.`val`)
                prev.next = current.next
            else 
                prev = current
            current = current.next
        }

        return dummy.next
    }
}
